src/main/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollector.java

```diff
@@ -115,6 +117,7 @@ public void collect(int doc) throws IOException {
                     collectedHitsPerSubQuery[i]++;
                     PriorityQueue<ScoreDoc> pq = compoundScores[i];
                     ScoreDoc currentDoc = new ScoreDoc(doc + docBase, score);
+                    maxScore = Math.max(score, maxScore);
```

Comment:
nit: May be move this close to score variable

Reply:
The motive here is to compare the score which is added in the scoredoc with the maxScore. That is the  reason I have added it after the currentDoc. I can change this condition to currentDoc.score()